x-pack/plugins/data_enhanced/server/search/session/monitoring_task.ts

```diff
@@ -25,12 +26,20 @@ interface SearchSessionTaskDeps {
   config: ConfigSchema;
 }
 
-function searchSessionRunner(core: CoreSetup, { logger, config }: SearchSessionTaskDeps) {
+function searchSessionRunner(
+  core: CoreSetup<DataEnhancedStartDependencies>,
+  { logger, config }: SearchSessionTaskDeps
+) {
   return ({ taskInstance }: RunContext) => {
     return {
       async run() {
         const sessionConfig = config.search.sessions;
-        const [coreStart] = await core.getStartServices();
+        const [coreStart, pluginsStart] = await core.getStartServices();
+        if (!sessionConfig.enabled) {
+          logger.info('Search sessions are disabled. Clearing task.');
+          await pluginsStart.taskManager.removeIfExists(SEARCH_SESSIONS_TASK_ID);
```

Comment:
@gmmorris, could you please review? (If this already hasn't been discussed somewhere) 🙏

Reply:
oooh I do not know about this 😬 
What `removeIfExists` will do here is delete the SO that backs the task that is **currently running**.
I'd expect that to cause this task execution to fail....

I suspect you'll see that if you add an end to end test that verifies this behaviour.

What are we trying to achieve here?
Are you looking to respond reactively to a config change?
A better approach would be to do this from outside the task execution - subscribe to the config change and delete the task. And, in the task executor, skip the code path you don't want to execute if config is disabled but the task was rerun before you got a chance to remove it.

Reply:
This is what we try to achive: 

1. Assume Kibana is running with this feature ON 
2. Restart Kibana with the feature OFF
3. We want to be sure that we don't run the task when the feature is OFF

Currently, this is not happening in master.

Reply:
@gmmorris Updated.
Since changing this setting restarts the server, I'm now unschedulng the task on the `setupMonitoring` function.

Reply:
👍